common/src/service/IotHubConnectionString.cs

```diff
@@ -85,12 +67,12 @@ public string GetPassword()
             return password;
         }
 
-        public string GetAuthorizationHeader()
+        public override string GetAuthorizationHeader()
```

Comment:
The base class now only has a abstract method and any auth implementation needs to provide the implementation. This is how we will be able to send a base class to all the clients but have specific way to get the tokens based on the auth type.